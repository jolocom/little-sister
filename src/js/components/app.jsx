import React from 'react'
import Reflux from 'reflux'
import Radium from 'radium'
import {bankUri} from 'lib/fixtures'

import {Layout, Content} from 'components/layout'
import {
  Paper,
  AppBar,
  IconButton,
  Snackbar,
  FlatButton,
  Dialog} from 'material-ui'
import Badge from 'material-ui/Badge'
import NavigationMenu from 'material-ui/svg-icons/navigation/menu'

import getMuiTheme from 'material-ui/styles/getMuiTheme'

import JolocomTheme from 'styles/jolocom-theme'

import LeftNav from 'components/left-nav/nav.jsx'
import Profile from 'components/accounts/profile.jsx'
import Tour from 'components/tour.jsx'

import Loading from 'components/common/loading.jsx'

import AccountActions from 'actions/account'
import AccountStore from 'stores/account'
import ConfirmStore from 'stores/confirm'

import PinnedActions from 'actions/pinned'
import ConfirmActions from 'actions/confirm'
import ProfileActions from 'actions/profile'
import ProfileStore from 'stores/profile'

import SnackbarStore from 'stores/snackbar'

// A pathname is considered public if either "/" or if it starts
// with any of the following publicRoutes
const publicRoutes = [
  '/login',
  '/signup',
  '/forgot-password',
  '/change-password'
]

let App = React.createClass({

  mixins: [
    Reflux.connect(AccountStore, 'account'),
    Reflux.connect(ProfileStore, 'profile'),
    Reflux.connect(SnackbarStore, 'snackbar'),
    Reflux.connect(ConfirmStore, 'confirm')
  ],

  propTypes: {
    location: React.PropTypes.object,
    children: React.PropTypes.node,
    route: React.PropTypes.object
  },

  contextTypes: {
    router: React.PropTypes.object.isRequired
  },

  childContextTypes: {
    muiTheme: React.PropTypes.object,
    profile: React.PropTypes.any,
    account: React.PropTypes.object,
    username: React.PropTypes.string,
    searchActive: React.PropTypes.bool,
    location: React.PropTypes.object,
    route: React.PropTypes.object
  },

  getChildContext: function () {
    let {account, profile, searchActive} = this.state
    return {
      muiTheme: this.theme,
      profile,
      account,
      username: account && account.username, // backward compat
      searchActive,
      location: this.props.location,
      route: this.props.route
    }
  },

  getInitialState() {
    return {
      searchActive: false,
      searchQuery: null,
      showIndicator: false
    }
  },

  componentWillMount() {
    this.theme = getMuiTheme(JolocomTheme)
    AccountActions.login()
  },

  componentDidUpdate(prevProps, prevState) {
    let {username} = this.state.account

    if (prevState.account.username === undefined ||
      prevState.account.username !== username) {
      this.checkLogin()
    }
  },

  isPublicRoute(path = this.props.location.pathname) {
    return path === '/' ||
      publicRoutes.some((publicRoute) => path.indexOf(publicRoute) === 0)
  },

  checkLogin() {
    let {username, loggingIn} = this.state.account

    // session is still loading, so return for now
    if (username === undefined && loggingIn) {
      return
    }

    if (!username && !this.isPublicRoute()) {
      this.context.router.push('/login')
    } else if (username && this.isPublicRoute()) {
      this.context.router.push('/graph')
    }

    if (username) {
      ProfileActions.load()
    }
  },

  _handlePinnedTap() {
    PinnedActions.show()
  },

  _handleSearchTap() {
    this.refs.search.show()
    this.setState({searchActive: true})
  },

  _handleSearchChange(query) {
    this.setState({searchQuery: query})
  },

  _handleSearchSubmit() {
    let uri = `${bankUri}/${this.state.searchQuery}#this`
    this.context.router.push(`/graph/${encodeURIComponent(uri)}`)
    this.refs.search.hide()
  },

  _handleSearchHide() {
    this.setState({
      searchActive: false,
      searchQuery: null
    })
  },

  _handleChatTap() {
    this.context.router.push('/conversations')
  },

  showDrawer() {
    this.refs.leftNav.show()
  },

  _handleConfirmClose() {
    ConfirmActions.close()
    this.state.confirm.callback()
  },

  getStyles() {
    let styles = {
      container: {
        width: '100%',
        height: '100%',
        position: 'relative'
      },
      header: {
        zIndex: 5,
        backgroundColor: this.theme.appBar.color
      },
      bar: {
        boxShadow: 'none'
      },
      nav: {
        display: this.state.searchActive ? 'none' : 'block'
      },
      icon: {
        color: this.theme.appBar.textColor
      },
      filters: {
        width: '100%',
        height: '48px'
      },
      menuIcon: {
        marginTop: '-20px',
        position: 'relative',
        top: '-4px',
        cursor: 'pointer'
      },
      hamburgerBadge: {
        top: 10,
        right: 20,
        width: 15,
        height: 15,
        display: 'none'
      }
    }
    return styles
  },

  render() {
    const styles = this.getStyles()

    // @TODO render login screen when logging in, also makes sures child
    // components don't get rendered before any user data is available
    if (this.state.account.loggingIn && !this.isPublicRoute()) {
      return <Loading />
    }

    // Deactivating search until we get it working
    /* <IconButton
       iconClassName="material-icons"
       iconStyle={styles.icon}
       onTouchTap={this._handleSearchTap}>search</IconButton> */
    const nav = (
      <div>
        <IconButton
          iconClassName="material-icons"
          iconStyle={styles.icon}
          onTouchTap={this._handleChatTap}>chat</IconButton>
      </div>
    )

    // Deactivating search until we get it working
    /*
    (
      <GraphSearch
        ref="search"
        onChange={this._handleSearchChange}
        onSubmit={this._handleSearchSubmit}
        onHide={this._handleSearchHide} />
    )
    */
    const search = null

    // Deactivating the filters until we get them working
    // <GraphFilters style={styles.filters} showDefaults />
    const filters = null

    const confirmActions = [
      <FlatButton
        label="Cancel"
        primary
        onTouchTap={this._handleConfirmClose}
      />,
      <FlatButton
        label={this.state.confirm.primaryActionText}
        primary
        onTouchTap={this._handleConfirmClose}
      />
    ]

    return (
      <div style={styles.container}>
        {this.isPublicRoute() ? this.props.children : (
          <Layout>
            <Paper style={styles.header}>
              <AppBar
                title="Graph"
                iconElementRight={nav}
                style={styles.bar}
                iconElementLeft={
                  <Badge
                    badgeContent={''}
                    secondary
                    badgeStyle={styles.hamburgerBadge}>
                    <NavigationMenu
                      onTouchTap={this.showDrawer}
                      style={styles.menuIcon} />
                  </Badge>
                }
                onLeftIconButtonTouchTap={this.showDrawer} />
              {filters}
              {search}
            </Paper>
            <LeftNav ref="leftNav" />
            <Content>
              {React.Children.map(this.props.children, (el) => {
                return React.cloneElement(el, {
                  searchQuery: this.state.searchQuery
                })
              })}
            </Content>
            <Profile />
            <Tour />
          </Layout>
        )}

        <Snackbar
          open={this.state.snackbar.open}
          message={this.state.snackbar.message}
          action={this.state.snackbar.undo && 'undo'}
          onActionTouchTap={this.state.snackbar.undoCallback}
        />

        <Dialog
          actions={confirmActions}
          modal={false}
          open={this.state.confirm.open}
          onRequestClose={this.handleClose}
        >
          {this.state.confirm.message}
        </Dialog>
      </div>

    )
  }

})

export default Radium(App)
