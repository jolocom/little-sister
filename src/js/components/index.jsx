import React from 'react'
import Radium from 'radium'
import {RaisedButton} from 'material-ui'
import Carousel from 're-carousel'
import IndicatorDots from 'components/common/indicator-dots.jsx'

let Index = React.createClass({
  contextTypes: {
    router: React.PropTypes.object,
    username: React.PropTypes.string,
    muiTheme: React.PropTypes.object
  },

  componentWillMount() {
    const {account} = this.context
    if (account && account.webId) {
      this.context.router.push('/graph')
    }
  },

  getStyles() {
    let {muiTheme} = this.context

    let styles = {
      container: {
        width: '100%',
        height: '100%',
        position: 'relative',
        display: 'flex',
        flexDirection: 'column',
        backgroundColor: muiTheme.jolocom.gray4
      },
      logo: {
        fontSize: '18px',
        fontWeight: '400',
        textAlign: 'center',
        marginTop: '24px',
        textTransform: 'uppercase'
      },
      logoImg: {
        width: '32px',
        height: '32px',
        verticalAlign: 'middle'
      },
      intro: {
        flex: 1,
        textAlign: 'center'
      },
      slide: {
        backgroundColor: muiTheme.jolocom.gray4,
        padding: '24px',
        height: '100%',
        display: 'flex',
        flexDirection: 'column'
      },
      content: {
        flex: 1
      },
      img: {
        maxWidth: '100%'
      },
      title: {
        color: muiTheme.jolocom.gray1,
        fontSize: '28px',
        fontWeight: '300'
      },
      logoStartImg: {
        padding: '24px',
        maxWidth: '80%',
        width: '320px',
        margin: 'auto',
        marginTop: '20vh'
      },
      actions: {
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'stretch',
        padding: '24px',
        maxWidth: '80%',
        width: '260px',
        margin: 'auto'
      },
      signup: {
        marginBottom: '12px'
      },
      login: {
      }
    }

    return styles
  },

  render() {
    let styles = this.getStyles()

    return (
      <div style={styles.container}>

        {/** <div style={styles.logo}>
          <img src="/img/logo.png" style={styles.logoImg} /> Jolocom
        </div> **/}

        {/** <Carousel style={styles.intro} indicator={IndicatorDots}>
          <div style={styles.slide}>
            <h3 style={styles.title}>Own your data</h3>
            <div style={styles.content}>
              <img src="/img/slide-data.png" style={styles.img} />
            </div>
          </div>
          <div style={styles.slide}>
            <h3 style={styles.title}>Create your personal digital identity</h3>
            <div style={styles.content}>
              <img src="/img/slide-identity.png" style={styles.img} />
            </div>
          </div>
          <div style={styles.slide}>
            <h3 style={styles.title}>Own your data</h3>
            <div style={styles.content}>
              <img src="/img/slide-share.png" style={styles.img} />
            </div>
          </div>
          <div style={styles.slide}>
            <h3 style={styles.title}>Visualize your data</h3>
            <div style={styles.content}>
              <img src="/img/slide-visualize.png" style={styles.img} />
            </div>
          </div>
        </Carousel> **/}

        <div style={styles.intro}>
          <img
            src="/img/logo_littlesister_start.svg"
            style={styles.logoStartImg} />
        </div>

        <div style={styles.actions}>
          <RaisedButton
            secondary
            label="Sign up"
            style={styles.signup}
            onClick={this._handleSignup}
          />
          <RaisedButton
            label="Log in"
            style={styles.login}
            onClick={this._handleLogin}
          />
        </div>
      </div>
    )
  },

  _handleSignup() {
    this.context.router.push('/signup')
  },

  _handleLogin() {
    this.context.router.push('/login')
  }

})

export default Radium(Index)
