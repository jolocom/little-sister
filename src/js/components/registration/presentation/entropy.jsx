import React from 'react'
import Radium from 'radium'
import MaskedImage from './masked-image'
import {RaisedButton} from 'material-ui'

import {Container, Footer} from '../../structure'

import {theme} from 'styles'

const IMAGE_DATA_URL = 'img/entropy.jpg'

const STYLES = {
  container: {
    backgroundImage: 'url(img/img_seedcreation.svg)',
    backgroundPosition: 'center',
    backgroundSize: 'cover',
    position: 'relative',
    padding: 0,
    '@media screen and (min-width: 768px)': {
      padding: '0'
    }
  },
  header: {
    position: 'absolute',
    top: '60px',
    width: '100%',
    maxWidth: '320px',
    padding: '0 16px',
    pointerEvents: 'none',
    boxSizing: 'border-box'
  },
  title: {
    color: '#fff',
    fontSize: theme.textStyles.headline.fontSize,
    fontWeight: theme.textStyles.headline.fontWeight,
    margin: 0
  },
  subtitle: {
    color: '#fff',
    fontSize: theme.textStyles.screenHeader.fontSize,
    fontWeight: theme.textStyles.headline.fontWeight,
    margin: 0
  },
  img: {
    alignItems: 'center',
    width: '100%',
    height: '100%'
  },
  footer: {
    position: 'absolute',
    bottom: 0,
    width: 'auto',
    padding: '32px',
    boxSizing: 'border-box'
  }
}

const Entropy = (props) => {
  let header

  if (props.imageUncoveredPaths.length === 0) {
    header = (
      <header style={STYLES.header}>
        <h1 style={STYLES.title}>
          Hi {props.user}, for... <br /><br />
          <span style={STYLES.subtitle}>
            ...more security we need some random data.<br /><br />
          Please put your finger anywhere on the screen and draw on it randomly.
          </span>
        </h1>
      </header>
    )
  }

  return (
    <Container
      style={STYLES.container}
    >

      {header}

      <MaskedImage
        image={IMAGE_DATA_URL}
        style={STYLES.img}
        maskColor={theme.palette.textColor}
        uncoveredPaths={props.imageUncoveredPaths}
        uncovering={props.imageUncovering}
        onPointUncovered={props.onImagePointUncoverd}
        onUncoveringChange={props.onImageUncoveringChange}
      />

      <Footer style={STYLES.footer}>
        <RaisedButton
          label="NEXT STEP"
          disabled={!props.valid}
          secondary
          onClick={props.onSubmit} />
      </Footer>
    </Container>
  )
}

Entropy.propTypes = {
  imageUncovering: React.PropTypes.bool.isRequired,
  imageUncoveredPaths: React.PropTypes.any,
  user: React.PropTypes.string,
  onImagePointUncoverd: React.PropTypes.func.isRequired,
  onImageUncoveringChange: React.PropTypes.func.isRequired,
  onSubmit: React.PropTypes.func.isRequired,
  valid: React.PropTypes.bool
}

export default Radium(Entropy)
