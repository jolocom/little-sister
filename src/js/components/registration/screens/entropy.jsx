import React from 'react'
import { connect } from 'redux/utils'
import ImageMaskBuilder from 'lib/image-mask-builder'
import Presentation from '../presentation/entropy'

@connect({
  props: ['registration'],
  actions: [
    'registration:goForward',
    'registration:setMaskedImageUncovering',
    'registration:addEntropyFromDeltas'
  ],
  pure: false
})
export default class RegistrationEntropyScreen extends React.Component {
  static propTypes = {
    registration: React.PropTypes.object.isRequired,

    goForward: React.PropTypes.func.isRequired,
    addEntropyFromDeltas: React.PropTypes.func.isRequired,
    setMaskedImageUncovering: React.PropTypes.func.isRequired
  }

  constructor() {
    super()

    this.state = {
      count: 0,
      imageMask: new ImageMaskBuilder()
    }
  }

  handleUncoveringChange(uncovering) {
    this.props.setMaskedImageUncovering(uncovering)

    if (uncovering) {
      this.state.imageMask.startNewPath()
    } else {
      this.state.imageMask.endPath()
    }
  }

  handleUncoveredPoint(x, y) {
    this.state.imageMask.addPoint(x, y)
    this.setState({count: this.state.count + 1})
  }

  render() {
    return <Presentation
      imageUncoveredPaths={this.state.imageMask.paths}
      imageUncovering={this.props.registration.maskedImage.uncovering}
      valid={this.props.registration.passphrase.sufficientEntropy}
      user={this.props.registration.humanName.value}
      onImagePointUncoverd={(...args) => this.handleUncoveredPoint(...args)}
      onImageUncoveringChange={(val) => this.handleUncoveringChange(val)}

      onSubmit={this.props.goForward}
    />
  }
}
