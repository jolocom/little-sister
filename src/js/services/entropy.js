import sjcl from 'sjcl'

export default class EntropyService {
  constructor() {
    this.generator = new sjcl.prng(5) // eslint-disable-line new-cap
    // this.generator.startCollectors()
  }

  addFromDelta(d) {
    this.generator.addEntropy(d, 1, 'mouse')
  }

  getProgress() {
    return this.generator.getProgress()
  }

  isReady() {
    return this.generator.isReady()
  }

  getRandomNumbers(count) {
    return this.generator.randomWords(count).map(s => parseInt(s))
  }

  getRandomString(wordCount) {
    return this.generator.randomWords(wordCount).join('')
  }
}
