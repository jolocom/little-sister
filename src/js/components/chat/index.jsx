import React from 'react'
import Radium from 'radium'

import {AppBar, Tabs, Tab, Paper, IconButton} from 'material-ui'

import Dialog from 'components/common/dialog.jsx'
import {Layout, Content} from 'components/layout'

import GraphStore from 'stores/graph-store'

@Radium
export default class Chat extends React.Component {
  static contextTypes = {
    router: React.PropTypes.any,
    store: React.PropTypes.object
  }

  static propTypes = {
    children: React.PropTypes.node,
    location: React.PropTypes.object
  }

  constructor(props) {
    super(props)

    this.state = {
      activeTab: this.getActiveTab(props.location.pathname)
    }
  }

  getActiveTab(path) {
    let activeTab = 'chat'

    if (path === '/contacts') {
      activeTab = 'contacts'
    }

    return activeTab
  }

  componentWillUpdate(newProps) {
    if (this.props.location.pathname !== newProps.location.pathname) {
      this.setState({
        activeTab: this.getActiveTab(newProps.location.pathname)
      })
    }
  }

  close() {
    if (GraphStore.state.center == null) {
      this.context.router.push('/graph')
    } else {
      this.context.router.push('/graph/' +
        encodeURIComponent(GraphStore.state.center.uri))
    }
  }

  render() {
    const backIcon = (
      <IconButton
        iconClassName="material-icons"
        iconStyle={styles.icon}
        onTouchTap={this._handleBackTouchTap}
      >
        arrow_back
      </IconButton>
    )

    /*
    const searchIcon = (
      <IconButton
        iconClassName="material-icons"
        iconStyle={styles.icon}
      >
        search
      </IconButton>
    )
    */

    return (
      <Dialog id="chat" visible fullscreen>
        <Layout>
          <Paper>
            <AppBar
              title="Chat"
              style={styles.bar}
              iconElementLeft={backIcon}
              // iconElementRight={searchIcon}
            />
            <Tabs valueLink={{
              value: this.state.activeTab,
              requestChange: (tab) => this._handleTabsChange(tab)
            }}>
              <Tab label="Conversations" value="chat" />
              <Tab label="Contacts" value="contacts" />
            </Tabs>
          </Paper>
          <Content style={styles.conversationsContainer}>
            {this.props.children}
          </Content>
        </Layout>
      </Dialog>
    )
  }

  _handleTabsChange(tab) {
    this.setState({activeTab: tab})

    switch (tab) {
      case 'chat':
        this.context.router.push('/conversations')
        break
      case 'contacts':
        this.context.router.push('/contacts')
        break
    }
  }

  _handleBackTouchTap = () => {
    this.close()
  }
}

let styles = {
  bar: {
    boxShadow: 'none'
  },
  conversationsContainer: {
    overflowY: 'auto'
  }
}
