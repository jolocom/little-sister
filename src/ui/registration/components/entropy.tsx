import React from 'react'
import { Button } from 'react-native-material-ui'
import { StyleSheet, View, Dimensions } from 'react-native'
import { JolocomTheme } from 'src/styles/jolocom-theme'
import { Block, Container, CenteredText } from 'src/ui/structure/'
import { MaskedImageComponent } from 'src/ui/registration/components/maskedImage'
import I18n from 'src/locales/i18n'

interface Props {
  addPoint: (x: number, y: number) => void
  submitEntropy: () => void
  readonly progress: number
}

const viewWidth: number = Dimensions.get('window').width

// TODO FONT WEIGHT REFERENCE FROM STYLES
const styles = StyleSheet.create({
  mainContainer: {
    backgroundColor: JolocomTheme.primaryColorBlack,
    padding: 0,
  },
  footerButton: {
    position: 'absolute',
    bottom: '5%',
  },
  text: {
    position: 'absolute',
    top: '20%',
    paddingHorizontal: viewWidth / 15,
    backgroundColor: JolocomTheme.primaryColorBlack,
    fontSize: JolocomTheme.headerFontSize,
    fontFamily: JolocomTheme.contentFontFamily,
    fontWeight: '100',
    color: JolocomTheme.primaryColorSand,
  },
  buttonContainer: {
    width: 164,
    height: 48,
    borderRadius: 4,
    backgroundColor: JolocomTheme.primaryColorPurple,
  },
  buttonText: {
    paddingVertical: 15,
    fontFamily: JolocomTheme.contentFontFamily,
    color: JolocomTheme.primaryColorWhite,
    fontSize: JolocomTheme.headerFontSize,
    fontWeight: '100',
  },
})

export const EntropyComponent: React.SFC<Props> = props => {
  const { progress, submitEntropy, addPoint } = props

  const msg =
    progress === 0
      ? I18n.t('For security purposes, we need some randomness') +
        '. ' +
        I18n.t('Please tap the screen and draw on it randomly')
      : `${Math.trunc(progress * 100)} %`

  return (
    <Container style={styles.mainContainer}>
      <CenteredText style={styles.text} msg={msg} />
      <Block>
        <MaskedImageComponent addPoint={addPoint} />
      </Block>
      <View style={styles.footerButton}>
        {progress === 1 ? (
          <Button
            style={{
              container: styles.buttonContainer,
              text: styles.buttonText,
            }}
            upperCase={false}
            raised={true}
            text={I18n.t('Continue')}
            onPress={submitEntropy}
          />
        ) : null}
      </View>
    </Container>
  )
}
