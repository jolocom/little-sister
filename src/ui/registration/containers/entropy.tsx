import React from 'react'
import { connect } from 'react-redux'
import { registrationActions } from 'src/actions'
import { EntropyComponent } from 'src/ui/registration/components/entropy'
import { RootState } from 'src/reducers'
import { 
  EntropyGeneratorInterface,
  EntropyGenerator
} from 'src/lib/entropyGenerator'

interface ConnectProps {
  submit: (encodedEntropy: string) => void;
}

interface OwnProps { }

interface Props extends OwnProps, ConnectProps {}

interface State {
  isDrawn: boolean
  encodedEntropy: string
  sufficientEntropy: boolean
  entropyProgress: number
}

export class EntropyContainer extends React.Component<Props, State> {
  private entropyGenerator! : EntropyGeneratorInterface

  state = {
    isDrawn: false,
    encodedEntropy: '',
    entropyProgress: 0,
    sufficientEntropy: false
  }

  componentDidMount() {
    this.entropyGenerator = this.setUpEntropyGenerator()
  }

  private setUpEntropyGenerator() : EntropyGenerator {
    return new EntropyGenerator()
  }

  private addPoint = (x: number, y: number) : void => {
    this.entropyGenerator.addFromDelta(x)
    this.entropyGenerator.addFromDelta(y)
    this.setState({ entropyProgress: this.entropyGenerator.getProgress() })
    this.updateEntropyProgress()
  }

  private updateEntropyProgress = () : void => {
    if (!this.state.sufficientEntropy && this.state.entropyProgress === 1) {
      const encodedEntropy = this.generateRandomString()
      this.setState({ encodedEntropy })
      this.setState({ sufficientEntropy: true })
    }
  }

  private generateRandomString = () : string => {
    return this.entropyGenerator.generateRandomString(4)
  }

  private submitEntropy = () : void => {
    this.props.submit(this.state.encodedEntropy)
  }

  render() {
    return (
      <EntropyComponent
        addPoint={ this.addPoint }
        progress={ this.state.entropyProgress }
        submitEntropy={ this.submitEntropy }
      />
    )
  }
}

const mapStateToProps = (state: RootState) => {
  return {}
}

const mapDispatchToProps = (dispatch: (action: Function) => void) => {
  return {
    submit: (encodedEntropy: string) => 
      dispatch(registrationActions.submitEntropy(encodedEntropy))
  }
}

export const Entropy = connect(mapStateToProps, mapDispatchToProps)(EntropyContainer)
