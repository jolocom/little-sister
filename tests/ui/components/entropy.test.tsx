import React from 'react'
import { shallow } from 'enzyme'
import { EntropyComponent } from 'src/ui/registration/components/entropy'

describe('Entropy component', ()=> {
  const props = {
    addPoint: (x: number, y: number) => null,
    submitEntropy: () => null,
  }

  it('matches the snapshot when not drawn upon and insufficient entropy', () => {
    const rendered = shallow(<EntropyComponent {...props} progress={0} />)
    expect(rendered).toMatchSnapshot()
  })

  it('matches the snapshot when drawn upon and insufficient entropy', () => {
    const rendered = shallow(<EntropyComponent {...props} progress={0.5} />)
    expect(rendered).toMatchSnapshot()
  })

  it('matches the snapshot with when drawn upon and sufficient entropy', () => {
    const rendered = shallow(<EntropyComponent {...props} progress={1} />)
    expect(rendered).toMatchSnapshot()
  })
})
