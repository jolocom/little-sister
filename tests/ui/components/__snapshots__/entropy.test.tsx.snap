// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entropy component matches the snapshot when drawn upon and insufficient entropy 1`] = `
<Container
  style={
    Object {
      "backgroundColor": "black",
    }
  }
>
  <CenteredText
    msg="50 %"
    style={
      Object {
        "backgroundColor": "black",
        "color": "#ffefdf",
        "fontFamily": "TTCommons",
        "fontSize": 22,
        "fontWeight": "100",
        "position": "absolute",
        "top": "20%",
      }
    }
  />
  <Block>
    <MaskedImageComponent
      addPoint={[Function]}
    />
  </Block>
  <Component
    style={
      Object {
        "bottom": "5%",
        "position": "absolute",
      }
    }
  >
    <Button
      accent={false}
      disabled={true}
      icon={null}
      onLongPress={null}
      onPress={[Function]}
      primary={false}
      raised={true}
      style={
        Object {
          "container": Object {},
          "text": Object {},
        }
      }
      testID={null}
      text="Continue"
      upperCase={true}
    />
  </Component>
</Container>
`;

exports[`Entropy component matches the snapshot when not drawn upon and insufficient entropy 1`] = `
<Container
  style={
    Object {
      "backgroundColor": "black",
    }
  }
>
  <CenteredText
    msg="For security purposes, we need some randomness. Please tap the screen and draw on it randomly"
    style={
      Object {
        "backgroundColor": "black",
        "color": "#ffefdf",
        "fontFamily": "TTCommons",
        "fontSize": 22,
        "fontWeight": "100",
        "position": "absolute",
        "top": "20%",
      }
    }
  />
  <Block>
    <MaskedImageComponent
      addPoint={[Function]}
    />
  </Block>
  <Component
    style={
      Object {
        "bottom": "5%",
        "position": "absolute",
      }
    }
  >
    <Button
      accent={false}
      disabled={true}
      icon={null}
      onLongPress={null}
      onPress={[Function]}
      primary={false}
      raised={true}
      style={
        Object {
          "container": Object {},
          "text": Object {},
        }
      }
      testID={null}
      text="Continue"
      upperCase={true}
    />
  </Component>
</Container>
`;

exports[`Entropy component matches the snapshot with when drawn upon and sufficient entropy 1`] = `
<Container
  style={
    Object {
      "backgroundColor": "black",
    }
  }
>
  <CenteredText
    msg="100 %"
    style={
      Object {
        "backgroundColor": "black",
        "color": "#ffefdf",
        "fontFamily": "TTCommons",
        "fontSize": 22,
        "fontWeight": "100",
        "position": "absolute",
        "top": "20%",
      }
    }
  />
  <Block>
    <MaskedImageComponent
      addPoint={[Function]}
    />
  </Block>
  <Component
    style={
      Object {
        "bottom": "5%",
        "position": "absolute",
      }
    }
  >
    <Button
      accent={false}
      disabled={false}
      icon={null}
      onLongPress={null}
      onPress={[Function]}
      primary={false}
      raised={true}
      style={
        Object {
          "container": Object {},
          "text": Object {},
        }
      }
      testID={null}
      text="Continue"
      upperCase={true}
    />
  </Component>
</Container>
`;
